### DISCUSSION TITLE
how to create custom controller with OnActionExecuting override method in orchardcore cms project?

### DISCUSSION THREAD

**Discussion by MRSheikh** [OrchardCMS/OrchardCore#12562] [Q&A]
hi.
I want to create 
`public override void OnActionExecuting(ActionExecutingContext filterContext){}`
method for my Orchardcore cms project. 
I created the Controller folder and `MyCmsController.cs` inside it. but OnActionExecuting is not firing. 
how to I implement OnActionExecuting for my cms??

**Comment by Skrypt** [comment]
You can use a class that inherits from ActionFilterAttribute.

**Comment by lampersky** [comment]
@MRSheikh 
I've just tried and there must be something wrong with your code, here it is my example controller, and as you can see I was able to add breakpoint, which is getting hit, when I'm visiting `https://localhost:5000/abc`

![image](https://user-images.githubusercontent.com/6403130/194134628-812c9575-e00d-44da-87cc-864e9189e498.png)
